pyquil/operator_estimation.py

```diff
@@ -479,3 +497,164 @@ def measure_observables(qc: QuantumComputer, tomo_experiment: TomographyExperime
                 expectation=np.asscalar(obs_mean),
                 stddev=np.asscalar(np.sqrt(obs_var)),
             )
+
+
+def diagonal_basis_commutes(pauli_a: PauliTerm, pauli_b: PauliTerm):
```

Comment:
Is "commute" the right word here?

Reply:
Good call out. "Commute" is actually a terrible word to use here. Even in the example of X Z and Z X given in the doc-string, the change-of-basis unitary matrices for their respective tpbs still commute with each other (even if those two operators are not diagonal in each others' tpbs). This function is basically just a natural generalization of the (private) function `_ops_diagonal_in_tpb(..)` to act on `PauliTerms` instead of just single-qubit operator strings. Therefore, I'm renaming this function to `terms_diagonal_in_tpb(..)`.